docs/conf.py

```diff
@@ -29,53 +27,178 @@
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
 extensions = [
+    "sphinx_automodapi.automodapi",
     "sphinx.ext.autodoc",
     "sphinx.ext.intersphinx",
-    "sphinx.ext.todo",
-    "sphinx.ext.coverage",
-    "sphinx.ext.inheritance_diagram",
-    "sphinx.ext.viewcode",
-    "sphinx.ext.napoleon",
-    "sphinx.ext.doctest",
+    "sphinx.ext.graphviz",
     "sphinx.ext.mathjax",
-    "sphinx_automodapi.automodapi",
-    "sphinx_automodapi.smart_resolver",
+    "sphinx.ext.napoleon",
+    "sphinx.ext.todo",
+    "nbsphinx",
+    "sphinx_copybutton",
+    "sphinx_gallery.load_style",
+    "IPython.sphinxext.ipython_console_highlighting",
+    "sphinx_changelog",
+    "sphinxcontrib.bibtex",
+    "hoverxref.extension",
 ]
 
+bibtex_bibfiles = ["bibliography.bib"]
+bibtex_default_style = "plain"
+bibtex_reference_style = "author_year"
+bibtex_cite_id = "{key}"
+
 # Add any paths that contain templates here, relative to this directory.
 # templates_path = ['_templates']
 
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
 # This pattern also affects html_static_path and html_extra_path.
-exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]
+exclude_patterns = [
+    ".DS_Store",
+    "_build",
+    "_links.rst",
+    "_substitutions.rst",
+    "Thumbs.db",
+]
 
 # The suffix(es) of source filenames.
 # You can specify multiple suffix as a list of string:
 source_suffix = ".rst"
 
-# The master toctree document.
-master_doc = "index"
+# The root toctree document.
+root_doc = "index"
 
 # -- Options for intersphinx extension ---------------------------------------
 
 intersphinx_mapping = {
-    "python": ("https://docs.python.org/3", None),
+    "astropy": ("https://docs.astropy.org/en/stable/", None),
+    "ndcube": ("https://docs.sunpy.org/projects/ndcube/en/stable/", None),
     "numpy": ("https://numpy.org/doc/stable/", None),
+    "python": ("https://docs.python.org/3", None),
     "scipy": ("https://docs.scipy.org/doc/scipy/reference/", None),
-    "pandas": ("https://pandas.pydata.org/pandas-docs/stable/", None),
-    "astropy": ("https://docs.astropy.org/en/stable/", None),
     "sunpy": ("https://docs.sunpy.org/en/stable/", None),
-    "ndcube": ("https://docs.sunpy.org/projects/ndcube/en/stable/", None),
 }
```

Comment:
Intersphinx is the Sphinx extension that lets us link to the documentation for other projects.  In our docs we can do something like...
```
This function returns a `~numpy.ndarray`.
```